pkg/devfile/adapters/kubernetes/component/adapter.go

```diff
@@ -282,6 +282,25 @@ func (a Adapter) createOrUpdateComponent(componentExists bool, ei envinfo.EnvSpe
 
 	kclient.AddBootstrapSupervisordInitContainer(podTemplateSpec)
 
+	// if there are preStart events, add them as init containers to the podTemplateSpec
+	preStartEvents := a.Devfile.Data.GetEvents().PreStart
+	if len(preStartEvents) > 0 {
+		var eventCommands []string
+		commandsMap := a.Devfile.Data.GetCommands()
+		containersMap := utils.GetContainersMap(containers)
+
+		for _, event := range preStartEvents {
+			eventSubCommands := common.GetCommandsFromEvent(commandsMap, strings.ToLower(event))
+			eventCommands = append(eventCommands, eventSubCommands...)
+		}
+
+		klog.V(4).Infof("PreStart event commands are: %v", strings.Join(eventCommands, ","))
+		utils.AddPreStartEventInitContainer(podTemplateSpec, commandsMap, eventCommands, containersMap)
+		if len(eventCommands) > 0 {
+			log.Successf("PreStart commands have been added to the component: %s", strings.Join(eventCommands, ","))
```

Comment:
Maybe let's print this string only when the component doesn't exist on the cluster.

Reply:
So I thought about that, and we shouldn't necessarily do it. Because init containers also execute when a deployment is restarted and that happens when container component values change in devfile, in between odo push. And currently in odo, we have no way of tracking if values changed between odo push since we dont track devfile data. Any other suggestions?

Reply:
I was thinking of something simpler like using the value of `componentExists` but as you said, that won't cover the case of a devfile data change. Let's leave this for now.